src/MongoDB.Driver/Linq/Expressions/FieldExpression.cs

```diff
@@ -60,6 +61,8 @@ public string FieldName
             get { return _fieldName; }
         }
 
+        public string OutOfCurrentScopePrefix { get; set; }
```

Comment:
Ok

Reply:
The reasons are:

1. It is helpful if all the members that hold state are listed together so that when reading the class one can understand what state an object holds by looking at one place
2. Any time you add any validation code to a property setter (null checks etc..) you need a backing field
3. You often need fields that aren't necessarily exposed publicly via properties

Since one is likely to end up with backing fields anyway for all but the simplest classes let's just always use them.

In the case of read-only properties manually adding an explicit backing field is one line of code.

Definitely let's never mix backing fields with automatic properties because then it's very easy to misinterpret what state an object can hold.

Reply:
Ok